src/btree/bt_cursor.c

```diff
@@ -512,12 +512,35 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
     return (0);
 }
 
+/*
+ * __wt_btcur_reposition --
+ *     Try to evict the page and reposition the cursor on the saved key.
+ */
+int
+__wt_btcur_reposition(WT_CURSOR_BTREE *cbt, bool release_page)
+{
+    WT_CURSOR *cursor;
+    WT_SESSION_IMPL *session;
+
+    cursor = &cbt->iface;
+    session = CUR2S(cbt);
+
+    if (release_page && session->txn->isolation == WT_ISO_SNAPSHOT) { // &&
+    //   !WT_SESSION_IS_CHECKPOINT(session) &&
+    //   F_ISSET_ATOMIC_16(cbt->ref->page, WT_PAGE_FORCE_EVICTION)) {
+        WT_RET(__wt_cursor_localkey(cursor));
+        WT_RET(__cursor_reset(cbt, true));
+        WT_RET(__wt_btcur_search(cbt, false));
```

Comment:
We should panic here is we cannot get the position back.

Reply:
Updated in [b7c4910](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/b7c4910805456568ee64adda417cacac2776df07)